src/components/ChoroplethMap.js

```diff
@@ -0,0 +1,161 @@
+import React, { Component } from 'react';
+import update from 'react/lib/update'
+import { connect } from 'react-redux';
+import PlotlyGraph from './Plot.js';
+import Select from 'react-select';
+import 'react-select/dist/react-select.css';
+import { fetchDataIfNeeded, riskAndDateAreSelected } from '../actions/ChoroplethMapActions';
+
+
+export class ChoroplethMap extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      graphID: 'ChoroplethMap',
+      data: [{
+        type: 'choropleth',
+        locationmode: 'country names',
+        locations: [],
+        z: [],
+        autocolorscale: false,
+        colorbar: {
+          thickness: 10,
+          x: 0.8
+        },
+        zmin: 0,
+        zmax: 8200000,
+        colorscale: [
+          [0, 'rgb(231, 251, 245)'],
+          [0.2, 'rgb(162, 239, 218)'],
+          [0.4, 'rgb(115, 231, 199)'],
+          [0.6, 'rgb(46, 219, 172)'],
+          [0.8, 'rgb(0, 212, 154)'],
+          [1, 'rgb(0, 174, 127)']
+        ]
+      }],
+      layout: {
+        height: 500,
+        title: 'Choropleth Map',
+        geo: {
+          showcountries: true,
+          projection: {
+              type: 'equirectangular'
+          }
+        }
+      },
+      dateToShow: this.props.view.date,
+      riskToShow: this.props.view.risk
+    }
+
+    this.handleChangeRisk = this.handleChange.bind(this, 'riskToShow')
+    this.handleChangeDate = this.handleChange.bind(this, 'dateToShow')
+    this.handleSubmit = this.handleSubmit.bind(this)
+  }
+
+
+  handleChange(idx, object) {
+    this.setState({
+      [idx]: object.value
+    })
+  }
+
+
+  handleSubmit(event) {
+    event.preventDefault()
```

Comment:
Let's get rid of submit and call appropriate actions on handleChangeRisk and  handleChangeDate